### PR TITLE
Channel discountCode integration tests refactor

### PR DESCRIPTION
**Pull request by barbara79**
1. code and add tests that cover the created code. Your code should be warning-free.
   * [ ] tests existing
1. stick to PSR-2 and and don't reformat existing code.
   * [ ] code style check succesful

### CODE REVIEW

File: tests/integration/CartDiscount/CartDiscountUpdateRequestTest.php [hunk 1]
```diff
@@ -357,23 +357,25 @@ function (CartDiscount $cartDiscount) use ($client) {
     public function testSetKey()
     {
         $client = $this->getApiClient();
+        $key = 'test-' . $this->getTestRun() . '-foo';
```
Comment (barbara79) [inline tests/integration/CartDiscount/CartDiscountUpdateRequestTest.php]: not sure if should I leave "$this->getTestRun()" or replace with something else

File: tests/integration/Type/TypeFixture.php [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+use Commercetools\Core\Model\Common\LocalizedString;
+use Commercetools\Core\Model\Type\FieldDefinition;
+use Commercetools\Core\Model\Type\FieldDefinitionCollection;
+use Commercetools\Core\Model\Type\StringType;
+use Commercetools\Core\Model\Type\Type;
+use Commercetools\Core\Model\Type\TypeDraft;
+use Commercetools\Core\Request\Types\TypeCreateRequest;
+use Commercetools\Core\Request\Types\TypeDeleteRequest;
+
+class TypeFixture extends ResourceFixture
```
Comment (barbara79) [inline tests/integration/Type/TypeFixture.php]: copied from the PR https://github.com/commercetools/commercetools-php-sdk/pull/524

File: tests/integration/CartDiscount/CartDiscountUpdateRequestTest.php (hunk 1 above)
Comment (barbara79) [reply tests/integration/CartDiscount/CartDiscountUpdateRequestTest.php]: and if we replace it with something like ?
$description = Uuid::uuidv4() . 'new-description';

Comment (jenschude) [reply tests/integration/CartDiscount/CartDiscountUpdateRequestTest.php]: If replacing it better use `'test-foo-'.Uuid::uuidv4()`

File: tests/integration/Channel/ChannelCreateRequestTest.php [hunk 3]
```diff
@@ -3,47 +3,32 @@
...
+        $client = $this->getApiClient();
+
+        ChannelFixture::withDraftChannel(
+            $client,
+            function (ChannelDraft $draft) {
+                return $draft->setKey('myChannel');
+            },
+            function (Channel $channel) use ($client) {
+                $request = RequestBuilder::of()->channels()->query()
+                    ->where('key="' . $channel->getKey() . '"');
```
Comment (jenschude) [inline tests/integration/Channel/ChannelCreateRequestTest.php]:
```suggestion
                    ->where('key=:key', ['key' => $channel->getKey()]);
```

File: tests/integration/CartDiscount/CartDiscountUpdateRequestTest.php (hunk 1 above)
Comment (jenschude) [reply tests/integration/CartDiscount/CartDiscountUpdateRequestTest.php]: Or directly use the CartDiscountFixture::uniqueCartDiscountString()

File: tests/integration/Channel/ChannelCreateRequestTest.php (hunk 3 above)
Comment (barbara79) [reply tests/integration/Channel/ChannelCreateRequestTest.php]: right! ;)

**Pull request merged**